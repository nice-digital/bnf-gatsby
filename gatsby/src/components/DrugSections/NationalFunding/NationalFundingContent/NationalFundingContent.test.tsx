import { screen, render } from "@testing-library/react";

import {
	NationalFundingContent,
	type NationalFundingContentProps,
} from "./NationalFundingContent";

describe("NationalFundingContent", () => {
	const props: NationalFundingContentProps = {
		heading: "<strong>NICE</strong> decisions",
		slug: "nice-decisions-budesonide",
		decisions: [
			{
				fundingIdentifier: "NICE TA187",
				title: "Infliximab and adalimumab for Crohnâ€™s disease (May 2010)",
				url: "http://www.nice.org.uk/guidance/ta187",
				approvedForUse: "Recommended",
			},
			{
				fundingIdentifier: "NICE TA329",
				title:
					"Infliximab, adalimumab and golimumab for treating moderately to severely active ulcerative colitis after the failure of conventional therapy (February 2015)",
				url: "Infliximab, adalimumab and golimumab for treating moderately to severely active ulcerative colitis after the failure of conventional therapy (February 2015)",
				approvedForUse: "Recommended with restrictions",
			},
		],
		contentFor: "budesonide",
		contentForPrefix: "For",
	};

	beforeEach(() => {
		// eslint-disable-next-line testing-library/no-render-in-lifecycle
		render(<NationalFundingContent {...props} />);
	});

	it("should render section with accessible name", () => {
		expect(
			screen.getByRole("region", { name: "NICE decisions For budesonide" })
		).toBeInTheDocument();
	});

	it("should render heading 4 with heading content", () => {
		expect(screen.getByRole("heading", { level: 4 })).toHaveTextContent(
			"NICE decisions For budesonide"
		);
	});

	it("should render list decision list with accessible name", () => {
		expect(
			screen.getByRole("list", { name: "NICE decisions For budesonide" })
		).toBeInTheDocument();
	});

	it("should render list item per decision", () => {
		expect(screen.getAllByRole("listitem")).toHaveLength(2);
	});

	it("should render link per decision", () => {
		expect(screen.getAllByRole("link")).toHaveLength(2);
	});

	it("should render decision link to given URL", () => {
		expect(screen.getByText("NICE TA187")).toHaveAttribute(
			"href",
			"http://www.nice.org.uk/guidance/ta187"
		);
	});

	it("should match snapshot", () => {
		expect(
			screen.getByRole("region", { name: "NICE decisions For budesonide" })
		).toMatchSnapshot();
	});
});
