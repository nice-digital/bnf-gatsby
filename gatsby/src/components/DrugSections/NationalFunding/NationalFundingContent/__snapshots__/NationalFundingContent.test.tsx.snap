// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NationalFundingContent should match snapshot 1`] = `
<section
  aria-labelledby="nice-decisions-budesonide"
>
  <h4
    id="nice-decisions-budesonide"
  >
    <span>
      <strong>
        NICE
      </strong>
       decisions
    </span>
    <span
      class="visually-hidden"
    >
       For budesonide
    </span>
  </h4>
  <ul
    aria-labelledby="nice-decisions-budesonide"
    class="list--unstyled list"
  >
    <li>
      <article
        class="card"
        data-component="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="http://www.nice.org.uk/guidance/ta187"
            >
              NICE TA187
            </a>
          </p>
        </header>
        <p
          class="card__summary"
        >
          <span>
            Infliximab and adalimumab for Crohnâ€™s disease (May 2010)
          </span>
        </p>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Funding decision:
            </dt>
            <dd>
              Recommended
            </dd>
          </div>
        </dl>
      </article>
    </li>
    <li>
      <article
        class="card"
        data-component="card"
      >
        <header
          class="card__header"
        >
          <p
            class="card__heading"
          >
            <a
              href="Infliximab, adalimumab and golimumab for treating moderately to severely active ulcerative colitis after the failure of conventional therapy (February 2015)"
            >
              NICE TA329
            </a>
          </p>
        </header>
        <p
          class="card__summary"
        >
          <span>
            Infliximab, adalimumab and golimumab for treating moderately to severely active ulcerative colitis after the failure of conventional therapy (February 2015)
          </span>
        </p>
        <dl
          class="card__metadata"
        >
          <div
            class="card__metadatum"
          >
            <dt
              class="visually-hidden"
            >
              Funding decision:
            </dt>
            <dd>
              Recommended with restrictions
            </dd>
          </div>
        </dl>
      </article>
    </li>
  </ul>
</section>
`;
